packages/wallet/src/W3mFrameProvider.ts

```diff
@@ -297,6 +309,11 @@ export class W3mFrameProvider {
   public async connect(payload?: W3mFrameTypes.Requests['AppGetUserRequest']) {
     try {
       const chainId = payload?.chainId || this.getLastUsedChainId() || 1
+
+      if (payload?.preferredAccountType) {
+        await this.setPreferredAccount(payload.preferredAccountType as W3mFrameTypes.AccountType)
+      }
```

Comment:
wouldn't calling this before `getUser` cause it to fail if there's no cached user here
https://github.com/reown-com/secure-appkit/blob/develop/packages/sdk/src/core/FrameSdk/index.tsx#L844
?

Reply:
nope, `getUser` is called only after successful connection